addon/components/flexberry-file.js

```diff
@@ -436,156 +586,206 @@ export default FlexberryBaseComponent.extend({
-
-      var uploadData = _this.get('uploadData');
+    return new Ember.RSVP.Promise((resolve, reject) => {
+      this.set('uploadIsInProgress', true);
 
-      // Additional data.
-      var initialValue = _this.get('initialValue');
+      let uploadData = this.get('uploadData');
+      let initialValue = this.get('initialValue');
       if (!Ember.isNone(initialValue)) {
         uploadData.formData = {
           // Metadata about previously uploaded file.
-          previousFileDescription: _this.get('initialValue')
+          previousFileDescription: initialValue
         };
       }
 
-      uploadData.submit().done(function(result, textStatus, jqXhr) {
-        var value = jqXhr.responseText;
+      uploadData.submit().done((result, textStatus, jqXhr) => {
+        let value = jqXhr.responseText;
 
-        _this.set('value', value);
-        _this.set('initialValue', Ember.copy(value, true));
-        _this.set('uploadData', null);
+        this.set('value', value);
+        this.set('initialValue', Ember.copy(value, true));
+        this.set('uploadData', null);
 
-        _this.sendAction('uploadSuccess', {
+        this.sendAction('uploadSuccess', {
           uploadData: uploadData,
           response: jqXhr,
           value: value
         });
-        resolve(_this.get('jsonValue'));
-      }).fail(function(jqXhr, textStatus, errorThrown) {
-        var fileName = ' \'' + file.name + '\'';
-        var errorText = errorThrown ? ' (' + errorThrown + ')' : '';
-        var errorTitle = i18n.t('components.flexberry-file.upload-file-error-title');
-        var errorContent = i18n.t('components.flexberry-file.upload-file-error-message', { fileName: fileName, errorText: errorText });
-
-        var showModalDialogOnUploadError = _this.get('showModalDialogOnUploadError');
-        if (showModalDialogOnUploadError) {
-          _this.showErrorModalDialog.call(_this, errorTitle, errorContent);
-        }
-
-        _this.sendAction('uploadFail', {
+        resolve(this.get('jsonValue'));
+      }).fail((jqXhr, textStatus, errorThrown) => {
+        let errorContent = this.showUploadErrorModalDialog(file.name, errorThrown ? ' (' + errorThrown + ')' : '');
+        this.sendAction('uploadFail', {
           uploadData: uploadData,
           response: jqXhr,
-          value: _this.get('value')
+          value: this.get('value')
         });
         reject(new Error(errorContent));
-      }).always(function() {
-        _this.set('uploadIsInProgress', false);
+      }).always(() => {
+        this.set('uploadIsInProgress', false);
       });
     });
   },
 
   /**
-   * Method to download uploaded file.
+    Method to download previously uploaded file.
+
+    @method downloadFile
    */
-  downloadFile: function() {
-    var jsonInitialValue = this.get('jsonInitialValue');
-    var fileUrl = this.get('jsonInitialValue.fileUrl');
+  downloadFile() {
+    let fileName = this.get('jsonInitialValue.fileName');
```

Comment:
Ember's get method does it, so in case this.get('a.b.c.d.e.f') it will normally retirn undefined even if conponent's 'a' property is null or undefined.
If you mean that there is check for fileUrl, but no check for fileName, then i can say that it is unnecessary, fileName is used only to display download error message, and if fileUrl is defined then fileName is defined too.